qctrlopencontrols/dynamic_decoupling_sequences/predefined.py

```diff
@@ -638,26 +723,64 @@ def new_periodic_sequence(
 def new_walsh_sequence(
     duration=None, paley_order=None, pre_post_rotation=False, **kwargs
 ):
-    """
-    Creates Walsh Single Axis Sequence.
+    r"""
+    Creates the Walsh sequence.
 
     Parameters
     ---------
     duration : float
-        Total duration of the sequence. Defaults to None
+        Total duration of the sequence :math:`\tau`. Defaults to None.
     paley_order : int, optional
-        Defaults to 1. The paley order of the walsh sequence.
+        The paley order :math:`k` of the walsh sequence. Defaults to 1.
     pre_post_rotation : bool, optional
-        If True, a :math:`\\pi.2` rotation is added at the
+        If True, a :math:`X_{\pi/2}` rotation is added at the
         start and end of the sequence.
     kwargs : dict
-        Additional keywords required by
-        qctrlopencontrols.dynamical_decoupling_sequences.DynamicDecouplingSequence
+        Additional keywords required by DynamicDecouplingSequence.
 
     Returns
     -------
-    qctrlopencontrols.dynamical_decoupling_sequences.DynamicDecouplingSequence
-        Walsh (single-axis) sequence
+    DynamicDecouplingSequence
+        The Walsh sequence.
+
+    Notes
+    -----
+    The Walsh sequence is defined by the switching function :math:`y(t)` given by a
+    Walsh function. To define the Walsh sequence, we first introduce the Rademacher
+    function [#]_, which is defined as
+
+    .. math::
+        R_j(x) := {\rm sgn}\left[\sin(2^j \pi x)\right] \;, \quad\; x \in [0, 1]\;, \; j \geq 0 \;.
+
+    The :math:`j`-th Rademacher function :math:`R_j(x)` is thus a periodic square wave switching
+    :math:`2^{j-1}` times between :math:`\pm 1` over the interval :math:`[0, 1]`. The Walsh
+    function of Paley order :math:`k` is denoted :math:`{\rm PAL}_k(x)` and defined as
+
+    .. math::
+        {\rm PAL}_k(x) = \Pi_{j = 1}^m R_j(x)^{b_j} \;, \quad\; x \ in [0, 1] \;.
+
+    where :math:`(b_m, b_{m-1}, \cdots, b_1)_2` is the binary representation of :math:`k`.
```

Comment:
What's the `_2` for?

Reply:
copied from the internal doc. I think it means binary representation.

Reply:
OK, personally I don't think it adds much, but I'm fine with keeping it if you think it does. FWIW the wiki doesn't have it (https://docs.q-ctrl.com/wiki/control-library#dynamical-decoupling-sequences).

Reply:
yeah, I think it's better to just remove it as I have to go back to the doc to check what it means. Besides, we did say it's binary representation right after it.